### PR TITLE
D3ASIM-3243: Readded area_reconfigure_event for devices in UpdateArea…

### PR DESCRIPTION
**Pull request by hannesdiedrich**
…Event.

### CODE REVIEW

File: src/d3a/d3a_core/live_events.py [hunk 1]
```diff
@@ -55,19 +55,21 @@ def apply(self, area):
...
                 area.strategy = InfiniteBusStrategy(**self.area_params)
-            else:
-                return False
-            area.activate()
-            area.strategy.event_activate()
-            area.strategy.event_market_cycle()
-        else:
-            area.area_reconfigure_event(**self.area_params)
+                reactivate = True
+            if reactivate:
```
Comment (Hassan754) [inline src/d3a/d3a_core/live_events.py:67]: I think you have some extra spaces here

Comment (hannesdiedrich) [reply src/d3a/d3a_core/live_events.py:67]: Sorry, I dont understand. Should I remove indention here?

Comment (Hassan754) [reply src/d3a/d3a_core/live_events.py:67]: Ah sorry I saw it indented to the right

**Review by Hassan754 — APPROVED** [review]
LGTM

**Review by spyrostz — APPROVED** [review]
LGTM

**Review by faizan2590 — APPROVED** [review]
LGTM

File: src/d3a/d3a_core/live_events.py [hunk 2]
```diff
@@ -55,19 +55,21 @@ def apply(self, area):
         if area_type is not None and area_type != "Area":
```
Comment (fievelk) [inline src/d3a/d3a_core/live_events.py:55]: Not part of this PR, but I've seen it in several lines of the codebase so I'll ask. Are we specifically checking for `None` values, or are we checking for broader false values instead? I would usually prefer `if area_type and area_type != 'Area'`, to consider cases in which `area_type` is something like `False`, `""` or `[]`.

**Review by fievelk — APPROVED** [review]
Just a small general question unrelated to the specific PR. Besides that, LGTM!

File: src/d3a/d3a_core/live_events.py (hunk 2 above)
Comment (hannesdiedrich) [reply src/d3a/d3a_core/live_events.py:55]: Here we are actually checking for None. But you are right, there are cases, where these checks should test for   `False`, `""` or `[]`, too. PLease help us to make this consistent. Thanks!

**Pull request merged**